src/on_disk_cache.rs

```diff
@@ -366,12 +394,57 @@ impl Cache {
     fn base_dir(&self) -> &Path {
         let base_dir = self.tempdir.as_ref().map(TempDir::path);
 
-        #[cfg(all(feature = "on-disk-cache", not(windows)))]
-        return base_dir.unwrap_or(&CACHE_DIRECTORY);
+        #[cfg(feature = "on-disk-cache")]
+        {
+            // Use persistent cache directory if available
+            base_dir.unwrap_or(&CACHE_DIRECTORY)
+        }
 
-        #[cfg(any(not(feature = "on-disk-cache"), windows))]
-        #[allow(clippy::unwrap_used)]
-        base_dir.unwrap()
+        #[cfg(not(feature = "on-disk-cache"))]
+        {
+            // When on-disk cache is disabled, always use temporary directory
+            #[allow(clippy::unwrap_used)]
+            base_dir.unwrap()
+        }
+    }
+
+    #[cfg(all(feature = "on-disk-cache", not(windows)))]
+    fn ensure_cache_dir(&self) -> Result<()> {
+        if self.tempdir.is_none() {
+            create_dir_all(&*CACHE_DIRECTORY).with_context(|| {
+                format!(
+                    "failed to create cache directory: {}",
+                    CACHE_DIRECTORY.display()
+                )
+            })?;
+        }
+        Ok(())
+    }
+
+    #[cfg(all(feature = "on-disk-cache", windows))]
+    fn ensure_cache_dir(&self) -> Result<()> {
+        if self.tempdir.is_none() {
+            create_dir_all(&*CACHE_DIRECTORY).with_context(|| {
+                format!(
+                    "failed to create cache directory: {}",
+                    CACHE_DIRECTORY.display()
+                )
+            })?;
+        }
+        Ok(())
+    }
+
+    #[cfg(not(feature = "on-disk-cache"))]
+    fn ensure_cache_dir(&self) -> Result<()> {
+        Ok(())
+    }
+}
+
+#[cfg(all(feature = "on-disk-cache", not(windows)))]
+pub(crate) fn purge_cache_directory() {
+    use std::fs::remove_dir_all;
+    if let Err(err) = remove_dir_all(&*CACHE_DIRECTORY) {
+        eprintln!("Failed to remove cache directory: {err}");
```

Comment:
Can this be written in the style `remover_dir_all(..).with_context(..)`?

Just using `eprintln` will not cause an error to bubble up and will not set the exit code. I know it's pedantic, but I'd prefer the code be consistent this way.